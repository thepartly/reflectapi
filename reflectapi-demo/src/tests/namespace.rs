#[test]
fn test_namespace_with_dash() {
    async fn empty<S>(_s: S, _: reflectapi::Empty, _: reflectapi::Empty) -> reflectapi::Empty {
        reflectapi::Empty {}
    }

    assert_builder_snapshot!(reflectapi::Builder::<()>::new()
        .name("pet-orders")
        .route(empty, |b| {
            b.name("jobs-two.pet-orders.list-x").description("desc")
        }))
}

#[test]
fn test_conflicting_names() {
    #[derive(Debug, serde::Serialize, reflectapi::Output)]
    struct Foo {}

    async fn foos_get<S>(_s: S, _: usize, _: reflectapi::Empty) -> Foo {
        Foo {}
    }

    assert_builder_snapshot!(reflectapi::Builder::<()>::new()
        .name("foos")
        .route(foos_get, |b| b.name("foos.get"))
        .rename_types("reflectapi_demo::tests::namespace::Foo", "foos::Foo"))
}

// Regression for rust client codegen where field `T` would resolve to the type `T` instead of the
// type parameter `T`.
#[test]
fn test_generic_and_type_conflict() {
    #[derive(
        Debug, serde::Serialize, serde::Deserialize, reflectapi::Input, reflectapi::Output,
    )]
    pub struct T<I>(I);

    #[derive(
        Debug, serde::Serialize, serde::Deserialize, reflectapi::Input, reflectapi::Output,
    )]
    pub struct K<T> {
        t: T,
    }

    async fn get<S>(_s: S, k: K<T<()>>, _: reflectapi::Empty) -> K<T<()>> {
        k
    }

    assert_builder_snapshot!(reflectapi::Builder::<()>::new()
        .route(get, |b| b.name("get"))
        .rename_types("reflectapi_demo::tests::namespace::T", "T"))
}
