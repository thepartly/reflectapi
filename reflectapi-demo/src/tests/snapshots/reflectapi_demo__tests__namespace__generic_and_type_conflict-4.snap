---
source: reflectapi-demo/src/tests/namespace.rs
expression: "reflectapi::codegen::openapi::Spec::from(&schema)"
snapshot_kind: text
---
{
  "openapi": "3.1.0",
  "info": {
    "title": "",
    "description": "",
    "version": "1.0.0"
  },
  "paths": {
    "/get": {
      "description": "",
      "post": {
        "operationId": "get",
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "type": "object",
                "title": "reflectapi_demo.tests.namespace.K<T<std.tuple.Tuple0>>",
                "required": [
                  "t"
                ],
                "properties": {
                  "t": {
                    "type": "array",
                    "prefixItems": [
                      {
                        "$ref": "#/components/schemas/std.tuple.Tuple0"
                      }
                    ]
                  }
                }
              }
            }
          },
          "required": true
        },
        "responses": {
          "200": {
            "description": "200 OK",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "title": "reflectapi_demo.tests.namespace.K<T<std.tuple.Tuple0>>",
                  "required": [
                    "t"
                  ],
                  "properties": {
                    "t": {
                      "type": "array",
                      "prefixItems": [
                        {
                          "$ref": "#/components/schemas/std.tuple.Tuple0"
                        }
                      ]
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "std.tuple.Tuple0": {
        "description": "Unit type",
        "type": "null"
      }
    }
  }
}
